docs/source/tutorials/improving_ml_performance.ipynb

```diff
@@ -0,0 +1,3573 @@
+       "              interaction_constraints=None, learning_rate=None, max_bin=None,\n",
+       "              max_cat_threshold=None, max_cat_to_onehot=None,\n",
+       "              max_delta_step=None, max_depth=None, max_leaves=None,\n",
+       "              min_child_weight=None, missing=nan, monotone_constraints=None,\n",
+       "              multi_strategy=None, n_estimators=None, n_jobs=None,\n",
+       "              num_parallel_tree=None, objective='multi:softprob', ...)"
+      ]
+     },
+     "execution_count": 40,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "clean_clf = XGBClassifier(tree_method=\"hist\", enable_categorical=True, random_state=SEED)\n",
+    "clean_clf.fit(train_features, train_labels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2fe23e39-fe7b-4145-af55-c7fc1f245850",
+   "metadata": {},
+   "source": [
+    "**In a Jupyter environment, please rerun this cell to show the HTML representation or trust the notebook.\n",
+    "On GitHub, the HTML representation is unable to render, please try loading this page with nbviewer.org.**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1b0f83f8-07e6-4702-a39e-94336268bfef",
+   "metadata": {},
+   "source": [
+    "### Evaluate classification model with clean training on clean test data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "id": "dcaeda51-9b24-4c04-889d-7e63563594fc",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Accuracy of model fit to clean training data, measured on clean test data: 77.2%\n"
+     ]
+    }
+   ],
+   "source": [
+    "clean_preds = clean_clf.predict(test_features)\n",
+    "acc_clean = accuracy_score(test_labels, clean_preds)\n",
+    "print(\n",
+    "    f\"Accuracy of model fit to clean training data, measured on clean test data: {round(acc_clean*100,1)}%\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d2624388-ad39-4c88-88c3-51a224ad549a",
+   "metadata": {},
+   "source": [
+    "## 8. Hyperparameter Optimization for editing data issues"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "96e8e3fe-b15f-41e0-87dd-0efb786f2920",
+   "metadata": {},
+   "source": [
+    "We have made some basic edits to improve test performance, so now we will parameterize each one of these edits (eg. what fraction of each issue to delete) to automatically find the best combination of edits to achieve optimal test performance. \n",
+    "\n",
+    "We will use a basic hyperparameter-tuning approach to optimize over these edit-variants + model re-training on the edited datasets with our objective being test performance.\n",
+    "\n",
+    "In a real-world setting, this would ideally be done on cleaned validation data instead of test data, but we are simplifying the approach for this tutorial.\n",
```

Comment:
Agree the terminology is confusing, what's your suggestion?
Here's our defs:

cleaning = fixing Datalab detected issues (eg. relabeling data, dropping some data)

the point of this sentence is just to clarify we are being suboptimal ML practitioners here because we are going to start tuning some decisions against the test data.  This call out is primarily to show we are not ML noobs and know this is not strictly kosher

Reply:
Great, OK. I have two suggestions:

1. Let's use {train/test/validation} terminology which I think is well understood in the ML community. You fit a model with feature selection on the training set against the test set, and then perform hyperparameter tuning on the validation set. The main point of my comment was that we start this tutorial saying that validation set == test set, but then here we revert back to the validation set having a meaning that is distinct from the test set
2. I like "cleaning". But then data that is in a state of not being cleaned should be "dirty" or "unclean", not "noisy". It does get a bit confusing when we're telling the reader (OK, we start with messy data, and then we apply preprocessing steps. Take the noisy data and find errors in it, which we can decide to auto-apply. When we're complete with cleaning".... Basically we should try our best (it's hard, I know!) to define adjectives like you do in your comment above. Noisy was one that I felt was unnecessarily confusing. Let's use "dirty" since it also ties in with branding. Let's not use Noisy since that can mean many things (e.g. just a low signal to noise ratio, too many columns)

Reply:
Oh and I used the suggestion feature - thanks for recommending @mturk24 ! 

https://github.com/cleanlab/cleanlab/pull/1071#discussion_r1646922207